### PR TITLE
Remove gox, add support for s390x

### LINKED ISSUE #31: support cross compile for other arch
Would like to build cli binaries for:
* linux/ppc64le
* linux/s390x

But at the moment `gox` does [not support s390x](https://github.com/mitchellh/gox/blob/c9740af9c6574448fd48eb30a71f964014c7a837/platform.go#L38-L45).

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #31

Removes `gox` and `scripts/build/linux-cross`
Adds support for linux/s390x

cc @andrewhsu @cpuguy83

### CODE REVIEW

**Comment by dnephin** [comment]
ping, this should be an easy review

File: scripts/build/cross [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
 BUILDDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-echo "Building all binaries"
+echo "Building binaries for all platforms"
 SHELL=/bin/bash parallel ::: \
-    "$BUILDDIR/linux-cross" \
     "$BUILDDIR/windows" \
     "$BUILDDIR/osx" \
+    "GOARCH=amd64   $BUILDDIR/binary" \
```
Comment (cpuguy83) [inline scripts/build/cross]: Should this set GOOS?

Comment (dnephin) [inline scripts/build/cross]: I guess right now it assumes they it's always run in a linux container, but that's not necessarily the case. 

I've added `GOOS=linux`

**Comment by thaJeztah** [comment]
This didn't make it into 17.06.0 https://github.com/docker/docker-ce/blob/17.06/components/cli/scripts/build/cross#L7-L12

**Pull request merged**